src/components/cc-addon-admin/cc-addon-admin.types.d.ts

```diff
@@ -0,0 +1,35 @@
+export type AddonAdminState = AddonAdminStateLoaded | AddonAdminStateLoading | AddonAdminStateError | AddonAdminStateSaving;
+
+export interface AddonAdminStateLoaded {
+  type: 'loaded';
+  name: string;
+  tags: string[];
+}
+
+export interface AddonAdminStateDeleting {
+  type: 'deleting';
+  name: string;
+  tags: string[];
+}
+
+export interface AddonAdminStateUpdatingName {
+  type: 'updatingName';
+  name: string;
+  tags: string[];
+}
+
+export interface AddonAdminStateUpdatingTags {
+  type: 'updatingTags';
+  name: string;
+  tags: string[];
+}
+
+export interface AddonAdminStateLoading {
+  type: 'loading';
+}
+
+export interface AddonAdminStateError {
+  type: 'error';
+}
+
+export type AddonAdminStateSaving = AddonAdminStateDeleting | AddonAdminStateUpdatingName | AddonAdminStateUpdatingTags;
```

Comment:
nitpick: I was a bit confused at first because this line was at the bottom. Can we move it with the other `type` at the top.

Other order proposition:

* type AddonAdminState
* interface AddonAdminStateLoaded
* interface AddonAdminStateLoading
* interface AddonAdminStateError
* type AddonAdminStateSaving
* interface AddonAdminStateDeleting
* interface AddonAdminStateUpdatingName
* interface AddonAdminStateUpdatingTags